### ISSUE TITLE
Bars slightly below horizontal rule lines

### ISSUE THREAD

**Issue by Maximization** [Abhinandan-Kushwaha/react-native-gifted-charts#670]
Hi, great library! Thank you very much.

I've got an issue where the bar heights are slightly below the horizontal rule lines. In this example, I expect the bars for Tue, Wed, and Sun to overlap the 4 and 2 rule lines.

![iPhone SE (3rd generation) - 2024-06-20 at 14 37 29](https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/9811142/9f2df918-3bbc-4937-ade3-2249b592855b)

I found that [`bottom: -0.5` is set on the bar](https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/blob/2c1bcffd40dec4c193e04dbc4230dab0acd77ed6/src/BarChart/Animated2DWithGradient.tsx#L76), and when I change this to `bottom: 0.5` it neatly touches the line.

What would you suggest?

**Comment by Abhinandan-Kushwaha** [comment]
Can you please share the data that you are rendering through the chart? @Maximization

**Comment by Maximization** [comment]
```js
[
  { "label": "Mon", "value": 2.3333333333333335 },
  { "label": "Tue", "value": 4 },
  { "label": "Wed", "value": 2 },
  { "label": "Thu", "value": 0 },
  { "label": "Fri", "value": 0 },
  { "label": "Sat", "value": 0 },
  { "label": "Sun", "value": 4 }
]
```

**Comment by Abhinandan-Kushwaha** [comment]
Hi @Maximization 
The bar heights are actually correct in the image you shared. The tip of the bar touches the rule lines. This is how it is supposed to work.

You are probably using the prop `roundedTop` or `barBorderRadius`. Remove them and see the bars will be perfectly aligned with the rule lines.

If you want the rounded part to add an extra height to the bars, then you need to add some extra value to the data items.
Here's an example where I have added `0.3` to each data item-

<img width="376" alt="barsPadded" src="https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/20596944/2310dfdc-0fd4-4c03-9055-c812849a2d2a">

The code for the above chart is-
```js
const App = () => {
  const data = [
    {label: 'Mon', value: 2.3333333333333335},
    {label: 'Tue', value: 4},
    {label: 'Wed', value: 2},
    {label: 'Thu', value: 0},
    {label: 'Fri', value: 0},
    {label: 'Sat', value: 0},
    {label: 'Sun', value: 4},
  ];

  return (
    <BarChart
      data={data.map(item => ({...item, value: item.value ? item.value + 0.3 : 0}))}
      roundedTop
      noOfSections={4}
      maxValue={4}
      frontColor="skyblue"
      yAxisExtraHeight={20}
    />
  );
}
```

**Comment by Maximization** [comment]
@Abhinandan-Kushwaha Is the bar supposed to touch (and stay under) or go on top of the rule lines?

In most of the examples in the docs ([such as this one](https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/blob/master/demos/barLabelled.png)) the bars go on top of the rule lines.

> You are probably using the prop roundedTop or barBorderRadius. Remove them and see the bars will be perfectly aligned with the rule lines.

I've removed the radius completely and you can see the bar below the rule line.

![iPhone SE (3rd generation) - 2024-06-21 at 22 31 19](https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/9811142/2356fe85-0d81-4b73-8c22-7c0ecd888336)

> If you want the rounded part to add an extra height to the bars, then you need to add some extra value to the data items.

This seems very hacky and would mess with the tooltip value as well.

Do you have another suggestion?

**Comment by Abhinandan-Kushwaha** [comment]
@Maximization The bar is supposed to touch and stay under the rules. Isn't this intuitive?

**Comment by Abhinandan-Kushwaha** [comment]
Hi @Maximization 
Sorry for the confusion and the late reply. It has now been fixed from version `1.4.15` onwards. 🎉
Please use the latest version of the library.
